kernel/cClosure.ml

```diff
@@ -1320,13 +1199,13 @@ let rec skip_irrelevant_stack info stk = match stk with
   let () = update m mk_irrelevant.mark mk_irrelevant.term in
   skip_irrelevant_stack info s
 
-let is_irrelevant_constructor infos (ind,_) = match infos.i_cache.i_mode with
-| Conversion -> Indset_env.mem ind infos.i_cache.i_env.irr_inds
-| Reduction -> false
+let is_irrelevant_constructor infos (ind,_) =
+  infos.i_cache.i_mode == Conversion &&
+  Indset_env.mem ind infos.i_cache.i_env.irr_inds
 
-let is_irrelevant_projection infos p = match infos.i_cache.i_mode with
-| Conversion -> not @@ Projection.Repr.relevant @@ Projection.repr p
-| Reduction -> false
+let is_irrelevant_projection infos p =
+  infos.i_cache.i_mode == Conversion &&
+  not (Projection.Repr.relevant (Projection.repr p))
```

Comment:
What happened here?

Reply:
Semantically, nothing, but this is more consistent with the changes I did to the `is_irrelevant` function.

Reply:
I think generally we prefer the previous style (matching on i_mode instead of using ==).
`is_irrelevant` specifically may have been nicer if written as 
~~~
let is_irrelevant info r = match info.i_cache.i_mode with
| Reduction -> false
| Conversion -> match r with
  | Sorts.Irrelevant -> true
  | Sorts.RelevanceVar q -> not (info.i_cache.i_sigma.qvar_relevant q)
  | Sorts.Relevant -> false
~~~